### ISSUE TITLE
Move logic out of `Listing` contract

### ISSUE THREAD

**Issue by wanderingstan** [OriginProtocol/origin-js#90]
Currently each instance of `Listing` contains a complete set of code for listing methods. This means higher gas costs all around.  The "correct" way to do this is to split the logic (ie all the methods) into a library contract, so that the actual `Listing` contact contains only data, and a new `ListingLibrary` contract contains the logic. 

@DanielVF has started work on this direction in his handling of `Purchase` contracts, here:
https://github.com/OriginProtocol/platform/commit/c4afd42966c029e68286cdaf85ca52aec2639b89

Once this is done, we'll want to do the same thing for all our contracts.

**Comment by cuongdo** [comment]
@wanderingstan Is this still a priority? I'd like to take this on.

**Comment by wanderingstan** [comment]
I believe this is already accomplisted via https://github.com/OriginProtocol/origin-js/pull/201 . Is this correct, @DanielVF ?

**Comment by cuongdo** [comment]
As far as I can tell, #201 separates storage and logic for `ListingsRegistry` not `Listing`.

**Comment by tyleryasaka** [comment]
This has *not* been done. Both listings an purchases can have code moved to a library.

**Comment by wanderingstan** [comment]
Doh! Yep, looks like its all you @cuongdo !

**Comment by cuongdo** [comment]
I think this is superseded by @tyleryasaka's listing and purchase upgradability work. What do you think @tyleryasaka ?

**Comment by DanielVF** [comment]
I agree, this issue is handled by the new listing contracts system.
